### ISSUE TITLE
Detect usage of deprecated provider_base method in resource

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#100]
### What category of cop is this?:

Please try to select as few as possible.

- [x] Deprecation - A cop that alerts to a deprecation in the Chef Infra Client
- [ ] Correctness - A cop that alerts when incorrect coding behavior is being used
- [ ] Sharing - A cop that alerts to missing functionality necessary for sharing cookboooks with others
- [ ] Style - A cop that alerts to a style best practice
- [ ] Simplification - A cop that alerts to ways code can be simplified through refactoring
- [ ] Effortless - A cop that alerts on code that must be resolved to move to the Effortless pattern
- [ ] Policyfiles - A cop that alerts on code that must be resolved to move to the Policyfiles pattern

### Describe the new cop:
The provider_base method was removed in Chef 13. We should detect anyone still using it.

### What it would trigger on:
```
Resource.provider_base 
```

### What it would autocorrect to if applicable
N/A:  See https://docs.chef.io/deprecations_custom_resource_cleanups.html for instructions

**Comment by tas50** [comment]
Fixed in #296
